airbyte-integrations/bases/standard-source-test/src/main/java/io/airbyte/integrations/standardtest/source/SourceComprehensiveTest.java

```diff
@@ -0,0 +1,147 @@
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package io.airbyte.integrations.standardtest.source;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.google.common.collect.Lists;
+import io.airbyte.db.Database;
+import io.airbyte.protocol.models.AirbyteMessage;
+import io.airbyte.protocol.models.AirbyteMessage.Type;
+import io.airbyte.protocol.models.CatalogHelpers;
+import io.airbyte.protocol.models.ConfiguredAirbyteCatalog;
+import io.airbyte.protocol.models.ConfiguredAirbyteStream;
+import io.airbyte.protocol.models.DestinationSyncMode;
+import io.airbyte.protocol.models.Field;
+import io.airbyte.protocol.models.JsonSchemaPrimitive;
+import io.airbyte.protocol.models.SyncMode;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.junit.jupiter.api.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public abstract class SourceComprehensiveTest extends SourceAbstractTest {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(SourceComprehensiveTest.class);
+
+  private final List<TestDataHolder> testDataHolders = new ArrayList<>();
+
+  /**
+   * Setup the test database. All tables and data described in the registered tests will be put there.
+   *
+   * @return configured test database
+   * @throws Exception - might throw any exception during initialization.
+   */
+  protected abstract Database setupDatabase() throws Exception;
+
+  /**
+   * Put all required tests here using method {@link #addDataTypeTestData(TestDataHolder)}
+   */
+  protected abstract void initTests();
+
+  @Override
+  protected void setupEnvironment(TestDestinationEnv environment) throws Exception {
+    setupDatabaseInternal();
+  }
+
+  /**
+   * The test checks that connector can fetch prepared data without failure.
+   */
+  @Test
+  public void testDataTypes() throws Exception {
+    ConfiguredAirbyteCatalog catalog = getConfiguredCatalog();
+    List<AirbyteMessage> allMessages = runRead(catalog);
+    final List<AirbyteMessage> recordMessages = allMessages.stream().filter(m -> m.getType() == Type.RECORD).collect(Collectors.toList());
+
+    recordMessages.forEach(msg -> LOGGER.debug(msg.toString()));
+    assertFalse(recordMessages.isEmpty(), "Expected records from source");
```

Comment:
I think we should assert proper values instead of just values being present

Reply:
Done